packages/rest/src/http/metadata.ts

```diff
@@ -430,6 +443,12 @@ export function createMetadataInfo(program: Program, options?: MetadataInfoOptio
     return true;
   }
 
+  function isOptional(property: ModelProperty, visibility: Visibility): boolean {
+    // Item flag is not a real visibility.
+    visibility &= ~Visibility.Item;
```

Comment:
In general the only deduping that happens if XxxSuffix is equivalent to Xxx (not transformed) then we use Xxx instead of XxxSuffix. If XxxSuffix1 and XxxSuffix2 are equivalent but neither are the same as Xxx then there will be duplication. It's very complicated to change this, unfortunately.

Reply:
Filed #1385

Reply:
OK cool, thanks a lot!